tests/integration/golang/aim/app/create_app_test.go

```diff
@@ -0,0 +1,100 @@
+//go:build integration
+package run
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/dao/models"
+	"github.com/G-Research/fasttrackml/tests/integration/golang/fixtures"
+	"github.com/G-Research/fasttrackml/tests/integration/golang/helpers"
+)
+
+type CreateAppTestSuite struct {
+	suite.Suite
+	client      *helpers.HttpClient
+	appFixtures *fixtures.AppFixtures
+	apps        []*models.App
+}
+
+func TestCreateAppTestSuite(t *testing.T) {
+	suite.Run(t, new(CreateAppTestSuite))
+}
+
+func (s *CreateAppTestSuite) SetupTest() {
+	s.client = helpers.NewAimApiClient(os.Getenv("SERVICE_BASE_URL"))
+
+	appFixtures, err := fixtures.NewAppFixtures(os.Getenv("DATABASE_DSN"))
+	assert.Nil(s.T(), err)
+	s.appFixtures = appFixtures
+
+	s.apps, err = s.appFixtures.CreateTestApps(context.Background(), 10)
+	assert.Nil(s.T(), err)
+}
+
+func (s *CreateAppTestSuite) Test_Ok() {
+	defer func() {
+		assert.Nil(s.T(), s.appFixtures.UnloadFixtures())
+	}()
+	tests := []struct {
+		name        string
+		requestBody models.App
+	}{
+		{
+			name: "CreateValidApp",
+			requestBody: models.App{
+				Type: "app-type",
+				State: models.AppState{
+					"app-state-key": "app-state-value",
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		s.T().Run(tt.name, func(T *testing.T) {
+
+			var resp models.App
+			err := s.client.DoPostRequest(
+				"/apps",
+				tt.requestBody,
+				&resp,
+			)
+			assert.Nil(s.T(), err)
+			assert.Equal(s.T(), "app-type", resp.Type)
```

Comment:
what about to check all other possible parameters? we probably have to check as much as possible.

Reply:
ok